hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestPartitionStatsIndexWithSql.scala

```diff
@@ -103,4 +117,154 @@ class TestPartitionStatsIndexWithSql extends HoodieSparkSqlTestBase {
       )
     }
   }
+
+  test("Test partition stats index on string type field with insert and file pruning") {
+    withTempDir { tmp =>
+      val tableName = generateTableName
```

Comment:
can we cover tests for both table types. 
and for non-partitioned. I understand there is just 1 partition. but wanted to have some sanity test

Reply:
made all tests run for both table types.. we already have a test for non-partitioned table with insert and udpate in `TestPartitionStatsIndex` with data source writes.